pymeasure/instruments/keithley/keithley2281S.py

```diff
@@ -0,0 +1,537 @@
+        ":CURR?",
+        ":CURR %g",
+        """Control the output current limit in Amps.""",
+        validator=strict_range,
+        values=_CURRENT_RANGE_PS,
+    )
+
+    ps_voltage_limit = Instrument.control(
+        ":VOLT:LIM?",
+        ":VOLT:LIM %g",
+        """Control the maximum voltage that can be set.""",
+        validator=strict_range,
+        values=_VOLTAGE_RANGE,
+    )
+
+    ps_power_supply_mode = Instrument.control(
+        ":FORM:ELEM:MODE?",
+        ":FORM:ELEM:MODE %s",
+        """Control which power supply mode to use.""",
+        validator=strict_discrete_set,
+        values={"CV", "CC", "OFF"},
+    )
+
+    ps_output_enabled = Instrument.control(
+        ":OUTP:STAT?",
+        ":OUTP:STAT %s",
+        """Control the output state.""",
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True,
+    )
+
+    ps_conc_nplc = Instrument.control(
+        ":SENS:CONC:NPLC?",
+        ":SENS:CONC:NPLC %g",
+        """
+        Control the number of power line cycles for current and voltage measurements.
+        The upper limit depends on the line frequency and is adapted at startup.
+        """,
+        validator=truncated_range,
+        values=_PLC_RANGE,
+    )
+
+    # Battery test (bt_*) commands, only applicable in battery test mode
+
+    bt_charge_voltage_setpoint = Instrument.control(
+        ":BATT:TEST:VOLT?",
+        ":BATT:TEST:VOLT %g",
+        """
+        Control the target voltage for battery tests.
+        If it is lower than terminal voltage, the battery gets discharged,
+        if it is higher than the terminal voltage the battery gets charged.
+        """,
+        validator=strict_range,
+        values=_VOLTAGE_RANGE,
+    )
+
+    bt_charge_voltage_limit = Instrument.control(
+        ":BATT:TEST:SENS:AH:VFUL?",
+        ":BATT:TEST:SENS:AH:VFUL %g",
+        """Control the charging target voltage for battery tests.""",
+        validator=strict_range,
+        values=_VOLTAGE_RANGE,
+    )
+
+    bt_charge_current_limit = Instrument.control(
+        ":BATT:TEST:SENS:AH:ILIM?",
+        ":BATT:TEST:SENS:AH:ILIM %g",
+        """Control the maximum charging current for battery tests.""",
+        validator=strict_range,
+        values=_CURRENT_RANGE_BT_BS,
+    )
+
+    bt_termination_current = Instrument.control(
+        ":BATT:TEST:CURR:END?",
+        ":BATT:TEST:CURR:END %g",
+        """Control the termination current for battery charging and discharging.""",
+        validator=truncated_range,
+        values=[0, 0.1],
+    )
+
+    bt_output_enabled = Instrument.control(
+        ":BATT:OUTP?",
+        ":BATT:OUTP %s",
+        """Control the output state.""",
+        validator=strict_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True,
+    )
+
+    bt_test_control = Instrument.setting(
+        ":BATT:TEST:SENS:AH:EXEC %s",
+        """Control the output state.""",
```

Comment:
It is a `setting`, so please use "Set":
```suggestion
        """Set the output state.""",
```

I do not understand, from the docstring, what do you set and what options do you have. I suggest to put the valid values into the docstring.

Reply:
Indeed, the Docstring is not enough here. There are two ways to enable the output in test-mode.
One is for (dis-)charging to a certain voltage (`bt_output_enabled`), the other is for actually recording the parameters for a battery-model.  
This is the property that triggers the output for characterization